### PR TITLE
another section for the logic chapter

### PR DESCRIPTION
**Pull request by avigad**
This builds on #11.

### CODE REVIEW

File: source/logic.rst [hunk 1]
```diff
@@ -0,0 +1,1223 @@
...
+
+We now have enough skills at our disposal to do some real mathematics.
+In Lean, we can represent a sequence :math:`s_0, s_1, s_2, \ldots` of
+real numbers as a function ``s : ℕ → ℝ``.
+Such a sequence is said to *converge* to a number :math:`a` if for every
+:math:`\varepsilon > 0` there is a point beyond which the sequence
+remains within :math:`\varepsilon` of :math:`a`,
+that is, there is a number :math:`N` such that for every
+:math:`n \ge N`, :math:`| s_n - a | < \varepsilon`.
+In Lean, we can render this as follows:
```
Comment (PatrickMassot) [inline source/logic.rst:900]: I would introduce the abbreviations `∀ ε > 0` very quickly, even before displaying `def converges_to`

File: source/logic.rst [hunk 2]
```diff
@@ -0,0 +1,1223 @@
...
+
+    variable (a : ℝ)
+
+    -- BEGIN
+    theorem converges_to_const : converges_to (λ x : ℕ, a) a :=
+    begin
+      intros ε εpos,
+      use 0,
+      intros n nge, dsimp,
+      rw [sub_self, abs_zero],
```
Comment (PatrickMassot) [inline source/logic.rst:998]: Maybe mention this will become easier at some point (after `simp` is explained).

File: source/logic.rst [hunk 3]
```diff
@@ -0,0 +1,1223 @@
...
+    theorem converges_to_add
+      (cs : converges_to s a) (ct : converges_to t b):
+    converges_to (λ n, s n + t n) (a + b) :=
+    begin
+      intros ε εpos, dsimp,
+      have ε2pos : 0 < ε / 2,
+      { linarith },
+      cases cs (ε / 2) ε2pos with Ns hs,
+      cases ct (ε / 2) ε2pos with Nt ht,
+      use max Ns Nt,
```
Comment (PatrickMassot) [inline source/logic.rst:1040]: This exercise needs lemmas about `max`.

File: source/logic.rst [hunk 4]
```diff
@@ -0,0 +1,1223 @@
...
+        rw [h, zero_mul] },
+      have acpos : 0 < abs c,
+        from abs_pos_of_ne_zero h,
+      sorry
+    end
+    -- END
+
+The next theorem is also independently interesting:
+it shows that a convergent sequence is eventually bounded
+in absolute value.
```
Comment (PatrickMassot) [inline source/logic.rst:1096]: You should warn readers that this stupid sounding lemma is suboptimal but sufficient for your purposes.

File: source/logic.rst [hunk 5]
```diff
@@ -0,0 +1,1223 @@
...
+    sorry
+
+    -- BEGIN
+    lemma aux (cs : converges_to s a) (ct : converges_to t 0) :
+      converges_to (λ n, s n * t n) 0 :=
+    begin
+      intros ε εpos, dsimp,
+      rcases exists_abs_le_of_converges_to cs with ⟨N₀, B, h₀⟩,
+      have Bpos : 0 < B,
+        from lt_of_le_of_lt (abs_nonneg _) (h₀ N₀ (le_refl _)),
```
Comment (PatrickMassot) [inline source/logic.rst:1146]: Maybe it's time to explain `linarith`.

File: source/logic.rst [hunk 6]
```diff
@@ -0,0 +1,1223 @@
...
+    ∀ ε, 0 < ε → ∃ N, ∀ n, N ≤ n → abs (s n - a) < ε
+    -- END
+
+.. TODO: reference to later chapter
+
+In a later chapter, we will see that mathlib has mechanisms
+for dealing with convergence in vastly more general terms,
+not only abstracting away particular features of the domain
+and codomain,
+but also abstracting over different *modes* of convergence.
```
Comment (PatrickMassot) [inline source/logic.rst:1223]: Is this "mode" standard terminology?

File: source/logic.rst (hunk 3 above)
Comment (avigad) [reply source/logic.rst:1040]: They did some things with `max` in the previous chapter, and at this stage I am hoping that they are starting to get comfortable with tab completion, searching mathlib, etc. But maybe I can add a hint.

File: source/logic.rst (hunk 2 above)
Comment (avigad) [reply source/logic.rst:998]: O.k. You should tell me when you think it is appropriate to introduce `simp`. Last time I tried, you told me that it was too early.

File: source/logic.rst (hunk 6 above)
Comment (avigad) [reply source/logic.rst:1223]: Not really. Maybe "different kinds of convergence"?

**Comment by avigad** [comment]
Good point on moving the convergence section to the end. I'll do that.

File: source/logic.rst (hunk 5 above)
Comment (avigad) [reply source/logic.rst:1146]: We use `linarith` in the previous chapter and also this one. But here is is not a big win: the line becomes `{ linarith [abs_nonneg (s N₀), h₀ N₀ (le_refl _)] }`.

**Pull request merged**